cmake/modules/CMakeFunctions.cmake

```
@@ -106,8 +106,10 @@ function(apply_patches repo_dir patches_dir base_revision target_branch ret)
         OUTPUT_QUIET
     )
     if(patches_needed EQUAL 128) # not a git repo
-        set(ret_not_git_repo 1)
-	elseif(patches_needed) # The target branch doesn't exist
+        set(${ret} True PARENT_SCOPE)
+        message(STATUS "[OPENCL-CLANG] ${repo_dir} is not a git repository")
+        return()
+    elseif(patches_needed EQUAL 1) # The target branch doesn't exist
```

do we need elseif after return?

---

there are 3 cases:
1. early return when repo is not a git repo
2. create branch and apply patches when target branch doesn't exist
3. switch to target branch when target branch exist.

return is in case1 and don't influence case2 and case3

---

Can we change elseif to if?
```suggestion
    endif()
    if(patches_needed EQUAL 1) # The target branch doesn't exist
```